### DISCUSSION TITLE
Saving and fetching from redis

### DISCUSSION THREAD

**Discussion by kaykhancheckpoint** [explosion/spaCy#9577] [Help: Coding & Implementations] [answered]
Hi friends,

I am creating an nlp named entity recognizer, i am saving this model to redis cache and then i am trying to retrieve from the redis cache.

```
async def create_nlp_model(entity_type):
    data = await redis.get(entity_type)
    if data is not None:
        print("from cache")
        nlp = English()
        return nlp.from_bytes(data)

    nlp = English()
    patterns = await get_patterns(entity_type)
    ruler = EntityRuler(nlp)
    ruler = nlp.add_pipe("entity_ruler")
    ruler.add_patterns(patterns)
    data = nlp.to_bytes()
    await redis.set(entity_type, nlp.to_bytes())

    return nlp
````    

```
import aioredis

import src.config as config

redis = aioredis.from_url(config.REDIS_HOST, encoding="utf-8")
```

However i have an issue the model returned from the cache does not work. By does not work i mean when it retrieves from the cache it does not recognise any entities. But the initial time (not using cache) it does recognize entities.

**Comment by polm** [comment, marked as answer]
Sorry you're having trouble with this. It's a little confusing, but just using `to_bytes`/`from_bytes` on a `Language` won't work - you need to do some other setup first to handle serializing the whole pipeline. This is documented [here](https://spacy.io/usage/saving-loading#pipeline).

**Reply by kaykhancheckpoint** [reply]
thanks got it working

**Comment by kaykhancheckpoint** [comment]


**Reply by polm** [reply]
That's correct.
